incubator/hnc/pkg/controllers/object_controller_new.go

```diff
@@ -299,31 +317,66 @@ func (r *ObjectReconcilerNew) delete(ctx context.Context, log logr.Logger, inst
 	if errors.IsNotFound(err) {
 		log.V(1).Info("The obsolete copy doesn't exist, no more action needed")
 		return nil
+	} else if err != nil {
+		r.setConditions(log, nil, inst, "delete", err)
+		return err
 	}
-	return err
+
+	return nil
 }
 
 func (r *ObjectReconcilerNew) write(ctx context.Context, log logr.Logger, inst, srcInst *unstructured.Unstructured) error {
 	// The object exists if CreationTimestamp is set. This flag enables us to have only 1 API call.
-	exist := inst.GetCreationTimestamp() != v1.Time{}
+	exist := inst.GetCreationTimestamp() != metav1.Time{}
 	ns := inst.GetNamespace()
 	inst = object.Canonical(srcInst)
 	inst.SetNamespace(ns)
 	metadata.SetLabel(inst, api.LabelInheritedFrom, srcInst.GetNamespace())
 	log.V(1).Info("Writing", "dst", inst.GetNamespace(), "origin", srcInst.GetNamespace())
 
 	var err error = nil
+	var op string
 	if exist {
 		err = r.Update(ctx, inst)
+		op = "update"
 	} else {
 		err = r.Create(ctx, inst)
+		op = "create"
 	}
 	if err != nil {
+		r.setConditions(log, srcInst, inst, op, err)
 		log.Error(err, "Couldn't write", "object", inst)
 	}
 	return err
 }
 
+func (r *ObjectReconcilerNew) setConditions(log logr.Logger, srcInst, inst *unstructured.Unstructured, op string, err error) {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+
+	gvk := inst.GetObjectKind().GroupVersionKind()
+	key := fmt.Sprintf("%s/%s/%s/%s/%s", gvk.Group, gvk.Version, gvk.Kind, inst.GetNamespace(), inst.GetName())
+	msg := fmt.Sprintf("Could not %s: %s", op, err.Error())
+	r.Forest.Get(inst.GetNamespace()).SetCondition(key, api.CannotUpdate, msg)
+	r.enqueueNamespace(log, inst.GetNamespace(), "Updating condition on propagated namespace: "+msg)
+	if srcInst != nil {
+		r.Forest.Get(inst.GetNamespace()).SetCondition(key, api.CannotPropagate, msg)
+		r.enqueueNamespace(log, inst.GetNamespace(), "Updating condition on source namespace: "+msg)
```

Comment:
Enqueue the `srcInst.GetNamespace()` here?

Reply:
Fixed